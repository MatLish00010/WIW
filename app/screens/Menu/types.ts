import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { HomeStackParamList } from '@myapp/navigation/HomeStack/types';

export type MenuScreenProps<Screen extends keyof HomeStackParamList> = NativeStackScreenProps<
  HomeStackParamList,
  Screen
>;
