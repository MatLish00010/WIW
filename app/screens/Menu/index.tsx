import { FC } from 'react';

import { UIViewContainer } from '@myapp/ui/View';
import { UIText } from '@myapp/ui/Text';
import { MenuScreenProps } from '@myapp/screens/Menu/types';

const Menu: FC<MenuScreenProps<'Menu'>> = () => {
  return (
    <UIViewContainer testID="menu-page">
      <UIText>Menu</UIText>
    </UIViewContainer>
  );
};

export default Menu;
