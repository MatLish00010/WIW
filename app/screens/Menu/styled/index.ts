import styled from 'styled-components/native';
import { palette, spacing } from '@myapp/ui/Theme';

export const MenuList = styled.FlatList`
  padding: ${spacing.M};
`;

export const MenuItem = styled.View`
  margin-bottom: ${spacing.XL};
  border-radius: 20px;
`;

export const MenuItemBtn = styled.TouchableOpacity`
  padding: ${spacing.XL};
  background-color: #00000c79;
  align-items: center;
  border-radius: 10px;
  color: ${palette.white};
`;

export const MenuItemBtnText = styled.Text`
  color: ${palette.white};
  font-weight: 500;
  text-transform: uppercase;
`;
