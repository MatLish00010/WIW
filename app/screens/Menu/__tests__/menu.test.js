import renderer from 'react-test-renderer';
import Menu from '@myapp/screens/Menu';
import renderWithTheme from '@myapp/utils/tests/renderWithTheme';

describe('Menu Test', () => {
  test('Page is correct', () => {
    const tree = renderer.create(renderWithTheme(<Menu />)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
