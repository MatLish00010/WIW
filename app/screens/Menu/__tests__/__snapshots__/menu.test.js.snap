// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Test Page is correct 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
        "paddingBottom": 32,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 40,
      },
    ]
  }
  testID="menu-page"
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
        },
      ]
    }
  >
    Menu
  </Text>
</View>
`;
